pkg/client/object.go

```diff
@@ -885,6 +894,110 @@ func (c *Client) objectPayloadRangeHashV2(ctx context.Context, p *RangeChecksumP
 	return res, nil
 }
 
+func (p *SearchObjectParams) WithContainerID(v *container.ID) *SearchObjectParams {
+	if p != nil {
+		p.cid = v
+	}
+
+	return p
+}
+
+func (p *SearchObjectParams) WithSearchFilters(v object.SearchFilters) *SearchObjectParams {
+	if p != nil {
+		p.filters = v
+	}
+
+	return p
+}
+
+func (c *Client) SearchObject(ctx context.Context, p *SearchObjectParams, opts ...CallOption) ([]object.ID, error) {
```

Comment:
I suggest to return `[]*object.ID` since we use pointers in most cases.

Reply:
Fixed